Source/WebCore/dom/Document.cpp

```diff
@@ -8080,6 +8082,57 @@ bool Document::resolveViewTransitionRule()
     return false;
 }
 
+void Document::reveal()
+{
+    if (!settings().crossDocumentViewTransitionsEnabled())
+        return;
+
+    if (m_hasBeenRevealed)
+        return;
+    m_hasBeenRevealed = true;
+
+    PageRevealEvent::Init init;
+
+    if (m_inboundViewTransitionParams && resolveViewTransitionRule())
+        init.viewTransition = ViewTransition::createInbound(*this, std::exchange(m_inboundViewTransitionParams, nullptr));
```

Comment:
Would be nice if part of these read more like the spec: https://drafts.csswg.org/css-view-transitions-2/#resolve-inbound-cross-document-view-transition

Reply:
Indeed it would, but there's a bunch I haven't implemented there. Will add a FIXME and work on it.